sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapter.java

```diff
@@ -132,7 +145,55 @@ public RootSDIAdapter addSDOI(String sdoName) {
         TSDI tsdi = new TSDI();
         tsdi.setName(sdoName);
         currentElem.getSDIOrDAI().add(tsdi);
-        return new RootSDIAdapter(this,tsdi);
+        return new RootSDIAdapter(this, tsdi);
+    }
+
+    public Optional<SclReportItem> updateDaiFromExtRef(List<TExtRef> tExtRefs) {
+        Optional<TExtRef> tExtRefMinOptional = tExtRefs.stream().min(Comparator.comparingInt(o -> Integer.parseInt(Objects.requireNonNull(Utils.extractField(o.getDesc(), "_", -1)))));
+
+        if (tExtRefMinOptional.isPresent() && extractDescSuffix(tExtRefMinOptional.get().getDesc()) == 1) {
+            TExtRef tExtRefMin = tExtRefMinOptional.get();
+            getDataAdapterByName("setSrcRef").updateVal(createInRefValNominalString(tExtRefMin));
+            if (tExtRefMin.isSetSrcCBName()) {
+                getDataAdapterByName("setSrcCB").updateVal(createInRefValTestString(tExtRefMin));
+            }
+
+            Optional<TExtRef> tExtRefMaxOptional = tExtRefs.stream().max(Comparator.comparingInt(o -> Integer.parseInt(Objects.requireNonNull(Utils.extractField(o.getDesc(), "_", -1)))));
+            if (tExtRefMaxOptional.isPresent() && extractDescSuffix(tExtRefMaxOptional.get().getDesc()) > 1) {
+                TExtRef tExtRefMax = tExtRefMaxOptional.get();
+                getDataAdapterByName("setTstRef").updateVal(createInRefValNominalString(tExtRefMax));
+                if (tExtRefMax.isSetSrcCBName()) {
+                    getDataAdapterByName("setTstCB").updateVal(createInRefValTestString(tExtRefMax));
+                }
+            }
+        } else {
+            return Optional.of(SclReportItem.warning(getXPath(), "The DOI %s can't be bound with an ExtRef".formatted(getXPath())));
+        }
+
+        return Optional.empty();
+    }
+
+    private Integer extractDescSuffix(String desc) {
+        return Integer.parseInt(Objects.requireNonNull(Utils.extractField(desc, "_", -1)));
+    }
+
+    private String createInRefValNominalString(TExtRef tExtRef) {
+        return tExtRef.getIedName() +
+                tExtRef.getLdInst() + "/" +
+                tExtRef.getPrefix() +
+                tExtRef.getLnClass().get(0) +
+                tExtRef.getLnInst() + "." +
+                tExtRef.getDoName();
```

Comment:
We make no guarantees that all of these fields are set. :flushed:
Going back a bit, only the `desc` field was previously verified as exist when this `ExtRef` was completed without filtering.